terraform-modules/routes/main.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_internet_gateway" "internet_gateway" {
+  vpc_id = "${var.vpc_id}"
+
+  tags = {
+    Name        = "${var.name}-internet-gateway"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_eip" "eip" {
+  depends_on = ["aws_internet_gateway.internet_gateway"]
+  vpc        = true
+
+  tags = {
+    Name        = "${var.name}"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_nat_gateway" "nat" {
+  depends_on    = ["aws_internet_gateway.internet_gateway"]
+  allocation_id = "${aws_eip.eip.id}"
+  subnet_id     = "${var.public_subnet_id}"
+
+  tags = {
+    Name        = "${var.name}"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_route_table" "private_route_table" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Name        = "${var.name}-private-route-table"
+    Environment = "${var.environment}"
+  }
+}
+
+resource "aws_route" "main_route" {
+  route_table_id         = "${var.main_route_table_id}"
+  destination_cidr_block = "${var.destination_cidr_block}"
+  gateway_id             = "${aws_internet_gateway.internet_gateway.id}"
```

Comment:
I don't think it is recommended practice to update the vpc main route table to allow traffic to the internet. Any Subnets created under the VPCs inherit the routes from the main routing table implicitly unless you associate a separate table to them. So it better to create a separate public table and explicitly assign it to public subnets. 
At least this is the case if done from aws console.